pkgs/os-specific/darwin/libtapi/default.nix

```diff
@@ -9,22 +9,36 @@ stdenv.mkDerivation {
     sha256 = "1a19h39a48agvnmal99n9j1fjadiqwib7hfzmn342wmgh9z3vk0g";
   };
 
+  sourceRoot = "source/src/llvm";
+
   nativeBuildInputs = [ cmake python3 ];
-  buildInputs = [ clang_6.cc ];
```

Comment:
i can't remember why clang_6 was needed here - but if everything builds correctly it sounds good to remove

Reply:
This project seems to be intended to be built from its source tree without an external clang. The source tree includes the parts of clang that are required. I didn't look at exactly how much is included or built, but the build is a lot shorter than a full llvm/clang build.

Taking clang out of the build inputs required adding `clangBasic` to the built targets to generate the required headers (compare to [upstream][clangBasic-dep]) and fixing the include directories to include the generated headers (compare to [upstream][header-fix]). The external clang was probably filling in these holes in the build system.

[header-fix]: https://github.com/tpoechtrager/apple-libtapi/blob/3cb307764cc5f1856c8a23bbdf3eb49dfc6bea48/build.sh#L59-L60
[clangBasic-dep]: https://github.com/tpoechtrager/apple-libtapi/blob/3cb307764cc5f1856c8a23bbdf3eb49dfc6bea48/build.sh#L71-L81